pkg/controller/podautoscaler/horizontal.go

```diff
@@ -845,9 +845,11 @@ func (a *HorizontalController) reconcileAutoscaler(ctx context.Context, hpaShare
 
 		logger.V(4).Info("Proposing desired replicas",
 			"desiredReplicas", metricDesiredReplicas,
+			"currentReplicas", currentReplicas,
 			"metric", metricName,
 			"timestamp", metricTimestamp,
-			"scaleTarget", reference)
+			"scaleTarget", reference,
```

Comment:
I checked and found that the modified logs below also use `reference := fmt.Sprintf("%s/%s/%s", hpa.Spec.ScaleTargetRef.Kind, hpa.Namespace, hpa.Spec.ScaleTargetRef.Name)` many times. Then you can directly use the following method at the top level of the call(Maybe under line 815), and the logs below will also have this key-value pair.

If other key-value pairs are used multiple times, you can also add them here.

```
if loggerV := logger.V(4); loggerV.Enabled() {
	logger = klog.LoggerWithValues(
		logger,
		"scaleTarget", reference,
		"currentReplicas", currentReplicas,
		...
	)
	ctx = klog.NewContext(ctx, logger)
}
```

Reply:
I think it's for logging, and is only utilized 3 times within this method. The rest are used in other methods. Switching to `loggerV` with `V(4)` might add complexity and reduce readability. It also can't be used for other verbose levels. I would suggest we keep it simple.

Reply:
It makes sense.